s3parq/session_helper.py

```diff
@@ -42,17 +57,26 @@ def make_db_session(self, **kwargs):
         self.Session = sessionmaker(bind=self.engine)
 
     def get_redshift_credentials(self):
-        client = self.boto_session.client(
-            'redshift',
-            region_name=self.region,
-            aws_access_key_id=self.aws_credentials.access_key,
-            aws_secret_access_key=self.aws_credentials.secret_key
-        )
-        temp_redshift_credentials = client.get_cluster_credentials(
-            DbUser=self.iam_user,
-            ClusterIdentifier=self.cluster_id,
-            AutoCreate=True,
-        )
+        if self._is_ec2():
```

Comment:
I noticed this is the second time you call this `self._is_ec2()`.  I think it might be better to call it in the `__init__` once and then set a flag called `_is_ec2`.  

Maybe you can change the function to be called `_set_is_ec2_flag()`

Does this make sense?  What you have now will technically work, so maybe leave it.  I'm pretty sure my example is only 1 call to the `169.254.169.254` URL instead of the two calls it currently makes, so it's not actually much of an improvement